### DISCUSSION TITLE
libX11: unset --disable-thread-safety-constructor

### DISCUSSION THREAD

**Discussion by Haagen-Dazs** [void-linux/void-packages#48804] [Q&A] [answered]
This commit [https://github.com/void-linux/void-packages/commit/51ab81ee8b546241ff382d07b464e257ee3ca2a6](https://github.com/void-linux/void-packages/commit/51ab81ee8b546241ff382d07b464e257ee3ca2a6) set `--disable-thread-safety-constructor` due to issues with xfsettingsd, which appears to be fixed upstream. See [https://gitlab.archlinux.org/archlinux/packaging/packages/libx11/-/commit/5bf321ab5544522cdb9e07cbcedb40a92afc6dfe](https://gitlab.archlinux.org/archlinux/packaging/packages/libx11/-/commit/5bf321ab5544522cdb9e07cbcedb40a92afc6dfe).
Is there any other reason to keep this?

**Comment by oreo639** [comment]
This is the actual upstream issue for reference: https://gitlab.freedesktop.org/xorg/lib/libx11/-/issues/157 (as stated there are more breakages than just xfsettingsd). This issue was closed in a patch in a feature branch (not in master) although allegedly the issue was fixed with glibc merging pthreads in to libc.so.

cc: @leahneukirchen

**Comment by Haagen-Dazs** [comment, marked as answer]
Closed by [https://github.com/void-linux/void-packages/commit/938b62dc5635cfaf4493377f9c77c5e54fa1ce39](https://github.com/void-linux/void-packages/commit/938b62dc5635cfaf4493377f9c77c5e54fa1ce39).
